### PR TITLE
[t3] introduce init Feed

### PR DESCRIPTION
**Pull request by rl3250**
### in this pr
What is Feed? Like Venmo, you can view your activity and your friend's activity in one place chronologically. 
feature: include promo texts for activities that involve business users.

### next steps
1. minor bug fixes?

### test plan
image

### CODE REVIEW

File: src/main/java/com/lgtm/easymoney/payload/FeedActivityRsp.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+/**
+ * transaction response.
+ */
+@AllArgsConstructor
+@Data
+@NoArgsConstructor
+public class FeedActivityRsp {
+  private Long fromUid;
+  private Long toUid;
+  private String amount;
```
Comment (rl3250) [inline src/main/java/com/lgtm/easymoney/payload/FeedActivityRsp.java:20]: to reviewers: need string to hide friend's activities(maybe in next pr)

File: src/main/java/com/lgtm/easymoney/services/impl/FeedServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import java.util.stream.Collectors;
+import net.sf.saxon.trans.SymbolicName;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+/**
+ * service for friendship.
+ * TODO after we have auth, move the inapplicable op checks to a HandlerInterceptor instead
```
Comment (rl3250) [inline src/main/java/com/lgtm/easymoney/services/impl/FeedServiceImpl.java:36]: forget to remove the comment :octocat:

File: src/main/java/com/lgtm/easymoney/services/impl/FeedServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  public FeedServiceImpl(UserService userService, FriendService friendService,
+                         TransactionService transactionService) {
+    this.userService = userService;
+    this.friendService = friendService;
+    this.transactionService = transactionService;
+  }
+
+  @Override
+  public List<Transaction> getFeedByUser(User u) {
+    // only return 20 latest and valid transaction
```
Comment (rl3250) [inline src/main/java/com/lgtm/easymoney/services/impl/FeedServiceImpl.java:62]: also in next pr

Comment (rl3250) [reply src/main/java/com/lgtm/easymoney/services/impl/FeedServiceImpl.java:62]: actually we can talk about this, i think feed should be time-sensitive and 20 sounds reasonable?

File: src/main/java/com/lgtm/easymoney/payload/FeedActivityRsp.java (hunk 1 above)
Comment (howieraem) [reply src/main/java/com/lgtm/easymoney/payload/FeedActivityRsp.java:20]: I think friend's activities should have amount = null

File: src/main/java/com/lgtm/easymoney/services/impl/FeedServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    // get user's activity
+    List<Transaction> userActivity = transactionService.getAllTransactionsWithUser(u,
+            List.of(TransactionStatus.TRANS_COMPLETE));
+    res.addAll(userActivity);
+    // get friends' activity
+    List<User> friends = friendService.getFriends(u);
+    for (User f : friends) {
+      List<Transaction> fs = transactionService
+              .getAllTransactionsWithUser(u, List.of(TransactionStatus.TRANS_COMPLETE));
+      res.addAll(fs);
```
Comment (howieraem) [inline src/main/java/com/lgtm/easymoney/services/impl/FeedServiceImpl.java:73]: Set friend's transaction amount to null?

**Review by howieraem — CHANGES_REQUESTED** [review]

File: src/main/java/com/lgtm/easymoney/services/impl/FeedServiceImpl.java (hunk 4 above)
Comment (howieraem) [reply src/main/java/com/lgtm/easymoney/services/impl/FeedServiceImpl.java:73]: and I think friend transactions is more appropriate if toUid is not PERSONAL?

**Review by howieraem — APPROVED** [review]

**Pull request merged**